test/src/test/java/org/corfudb/runtime/view/ManagementViewTest.java

```diff
@@ -1499,6 +1501,113 @@ public void verifyStateTransferNotHappenButMergeSucceeds() throws Exception {
         waitForLayoutChange(l -> l.getSegments().size() == 1, rt);
     }
 
+    /**
+     * This test verifies that partial state transfers when retried only transfer the delta
+     * of the address space.
+     * Setup: Layout - Write 11_000 entries to SERVER 0 and 1_000 entries to SERVERS 0 & 1.
+     * We then trigger the state transfer and force the transfer to fail on the last range write.
+     * Another transfer is then triggered and verified that only the remaining data is transferred.
+     */
+    @Test
+    public void verifyPartialStateTransferCompletionOnRetry() throws Exception {
+        corfuRuntime = null;
+        try {
+
+            ServerContext sc1 = new ServerContextBuilder()
+                    .setSingle(false)
+                    .setServerRouter(new TestServerRouter(SERVERS.PORT_0))
+                    .setPort(SERVERS.PORT_0).build();
+            addServer(SERVERS.PORT_0, sc1);
+
+            ServerContext sc2 = new ServerContextBuilder()
+                    .setMemory(false)
+                    .setSingle(false)
+                    .setLogPath(PARAMETERS.TEST_TEMP_DIR)
+                    .setServerRouter(new TestServerRouter(SERVERS.PORT_1))
+                    .setPort(SERVERS.PORT_1).build();
+            addServer(SERVERS.PORT_1, sc2);
+
+            getManagementServer(SERVERS.PORT_0).shutdown();
+            getManagementServer(SERVERS.PORT_1).shutdown();
+
+            final long writtenAddressesBatch1 = 11_000L;
+            final long writtenAddressesBatch2 = 1_000L;
+
+            Layout layout = new TestLayoutBuilder()
+                    .setEpoch(1L)
+                    .addLayoutServer(SERVERS.PORT_0)
+                    .addLayoutServer(SERVERS.PORT_1)
+                    .addSequencer(SERVERS.PORT_0)
+                    .addSequencer(SERVERS.PORT_1)
+                    .buildSegment()
+                    .setStart(0L)
+                    .setEnd(writtenAddressesBatch1)
+                    .buildStripe()
+                    .addLogUnit(SERVERS.PORT_0)
+                    .addToSegment()
+                    .addToLayout()
+                    .buildSegment()
+                    .setStart(writtenAddressesBatch1)
+                    .setEnd(-1)
+                    .buildStripe()
+                    .addLogUnit(SERVERS.PORT_0)
```

Comment:
Already did addLogUnit(SERVERS.PORT_0) at line 1546

Reply:
These are 2 different segments which are being created.
Your layout needs to look like this:
```
{
  "layoutServers": [
    "localhost:9000",
    "localhost:9001"
  ],
  "sequencers": [
    "localhost:9000",
    "localhost:9001"
  ],
  "segments": [
    {
      "replicationMode": "CHAIN_REPLICATION",
      "start": 0,
      "end": 11000,
      "stripes": [
        {
          "logServers": [
            "localhost:9000"
          ]
        }
      ]
    },
    {
      "replicationMode": "CHAIN_REPLICATION",
      "start": 11000,
      "end": -1,
      "stripes": [
        {
          "logServers": [
            "localhost:9000",
            "localhost:9001"
          ]
        }
      ]
    }
  ],
  "epoch": 1
}
```